param.go

```diff
@@ -92,6 +92,24 @@ var (
 	_ param = paramList{}
 )
 
+// Calls the provided function on all paramSingles in the given param tree.
+func forEachSimpleParam(param param, f func(paramSingle)) {
+	switch p := param.(type) {
+	case paramList:
+		for _, arg := range p.Params {
+			forEachSimpleParam(arg, f)
+		}
+	case paramSingle:
+		f(p)
+	case paramObject:
+		for _, field := range p.Fields {
+			forEachSimpleParam(field.Param, f)
+		}
+	default:
+		panic(fmt.Sprintf("unknown param type %T", param))
```

Comment:
Same comment here about the panic message: let's make it absolutely clear that this is a bug in dig, and that the user can't possibly fix it.